plugins/database/quota/src/org/apache/cloudstack/api/response/QuotaResponseBuilderImpl.java

```diff
@@ -0,0 +1,419 @@
+            }
+        }
+
+        if (quota_activity > 0 && quotaBalance.size() > 0) {
+            // order is desc last item is the start item
+            QuotaBalanceVO startItem = quotaBalance.get(quotaBalance.size() - 1);
+            QuotaBalanceVO endItem = quotaBalance.get(0);
+            resp.setStartDate(startDate);
+            resp.setStartQuota(startItem.getCreditBalance());
+            resp.setEndDate(endDate);
+            resp.setEndQuota(endItem.getCreditBalance().add(lastCredits));
+        } else if (quota_activity > 0) {
+            // order is desc last item is the start item
+            resp.setStartDate(startDate);
+            resp.setStartQuota(new BigDecimal(0));
+            resp.setEndDate(endDate);
+            resp.setEndQuota(new BigDecimal(0).add(lastCredits));
+        } else {
+            resp.setStartQuota(new BigDecimal(0));
+            resp.setEndQuota(new BigDecimal(0));
+        }
+        resp.setCurrency(QuotaConfig.QuotaCurrencySymbol.value());
+        resp.setObjectName("balance");
+        return resp;
+    }
+
+    @Override
+    public QuotaStatementResponse createQuotaStatementResponse(final List<QuotaUsageVO> quotaUsage) {
+        if (quotaUsage == null || quotaUsage.size() == 0) {
+            throw new InvalidParameterValueException("There is no usage data found for period mentioned.");
+        }
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        TransactionLegacy.open(TransactionLegacy.USAGE_DB).close();
+        QuotaStatementResponse statement = new QuotaStatementResponse();
+
+        HashMap<Integer, QuotaTariffVO> quotaTariffMap = new HashMap<Integer, QuotaTariffVO>();
+        List<QuotaTariffVO> result = _quotaTariffDao.listAll();
+
+        for (QuotaTariffVO quotaTariff : result) {
+            quotaTariffMap.put(quotaTariff.getUsageType(), quotaTariff);
+            // add dummy record for each usage type
+            QuotaUsageVO dummy = new QuotaUsageVO(quotaUsage.get(0));
+            dummy.setUsageType(quotaTariff.getUsageType());
+            dummy.setQuotaUsed(new BigDecimal(0));
+            quotaUsage.add(dummy);
+        }
+
+        Collections.sort(quotaUsage, new Comparator<QuotaUsageVO>() {
+            public int compare(QuotaUsageVO o1, QuotaUsageVO o2) {
+                if (o1.getUsageType() == o2.getUsageType())
+                    return 0;
+                return o1.getUsageType() < o2.getUsageType() ? -1 : 1;
+            }
+        });
+
+        List<QuotaStatementItemResponse> items = new ArrayList<QuotaStatementItemResponse>();
+        QuotaStatementItemResponse lineitem;
+        int type = -1;
+        BigDecimal usage = new BigDecimal(0);
+        BigDecimal totalUsage = new BigDecimal(0);
+        quotaUsage.add(new QuotaUsageVO());// boundary
+        QuotaUsageVO prev = quotaUsage.get(0);
+        // s_logger.info("createQuotaStatementResponse record count=" + quotaUsage.size());
```

Comment:
Please remove commented code -- it is cruft.  git provides the means to track changes in code including removed code.